### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'replace'

### ISSUE THREAD

**Issue by chuckwondo** [MAAP-Project/gedi-subsetter#41]
Some users encounter the following error during job execution, even after the job has been running for some time without issue:

```
AttributeError: 'NoneType' object has no attribute 'replace'
```

Here is the job ID for one such job (submitted by ): aa6e4581-0407-47f8-931d-555c5ad9e3b9

**Comment by chuckwondo** [comment]
I was able to reproduce the error within the ADE, using the following:

```plain
./subset.sh --aoi /projects/shared-buckets/abarenblitt/PA_Subset_Senegal.geojson --doi 10.5067/GEDI/GEDI02_B.002 --lat geolocation/lat_lowestmode --lon geolocation/lon_lowestmode --columns rh100 --query "l2a_quality_flag == 1 and l2b_quality_flag == 1 and sensitivity > 0.95"
```

After some debugging, the problem turns out to be a bug in the `maap-py` library. This occurs when a granule's metadata does NOT contain a download URL, so the Granule object constructed by `maap-py` contains a `None` value for its download URL, thus producing the error message when calling the `getData` method to download the file (within the `getData` method, the `replace` method is called on the download URL, under the assumption that the URL is a string, not None).

Fortunately, we can handle this on our end by simply filtering out any Granule objects from our CMR query where the Granules' download URLs are `None`.

**Comment by chuckwondo** [comment]
Fixed in v0.6.0, #39
